samples/99_zil_interpreter/app/parser.rb

```diff
@@ -0,0 +1,375 @@
+# Parser class uses the scanner to load from file or string buffer. Results are loaded into expressions.
+#
+# Usage:
+#   source = "<ZIL ()>"
+#   parser.parse_string(args, source)
+#   expr0 = parser.expressions[0]
+#
+#   OR
+#
+#   parser.parse_file(args, '/data/zil/zork1/', 'zork1.zil')
+#   expr0 = parser.expressions[0]
+#
+class Parser
+  attr_reader :expressions
+
+  def initialize
+    @expressions = []
+  end
+
+  def parse_file(args, work_dir, file_name)
+    @work_dir = work_dir # not sure if we need work_dir, the idea was to follow INSERT-FILEs inside the parser/scanner
+
+    file = args.gtk.read_file(@work_dir + file_name)
+    log "Opening '" + file_name + "' ..."
+
+    buffer = ""
+    file.each_line.each { |line|
+      line.each_char { |c| buffer << c }
+    }

```

Comment:
I wonder why are you reconstructing buffer from all characters of `file` ? Why not use `file` directly?

Reply:
This will be fixed in next commit.